containerd-shim-spin-v1/src/main.rs

```diff
@@ -126,17 +132,20 @@ impl Instance for Wasi {
             stdout: cfg.get_stdout().unwrap_or_default(),
             stderr: cfg.get_stderr().unwrap_or_default(),
             bundle: cfg.get_bundle().unwrap_or_default(),
+            shutdown_tx: Arc::new(Mutex::new(tx)),
```

Comment:
I don't think you need arc/mutex around channel?

Reply:
The channel is sent to a thread, so I need it to impl Send and Sync traits. Thus I used Mutex on it. I think I can remove Arc

Reply:
The reason a channel exists is to send data between threads.
I believe all you should need to do is pass a cloned channel end (and not reference the clone anywhere but after moving to the thread).

Reply:
It makes sense! Thanks, I will change that.

Reply:
rust channel Receiver cannot be `cloned`. Also see here: https://www.reddit.com/r/rust/comments/2lma84/why_cant_you_clone_a_receivert_like_you_can_a/. 

I created the channel in the `new` function and pass the receiver to another thread to wait for the shutdown signal - and this is why I need the `Receiver` to be `Send + Sync`.

Reply:
I actually changed the message passing implementation to using `condvar`, which it makes a bit more sense in this case. I only need a `kill` signal to shutdown my server.